collated/main/tby1994.md

```diff
@@ -0,0 +1,884 @@
+
+    private String commandMode;
+
+    public CommandMode() {
+        commandMode = "ab";
+    }
+    /**
+     * Validates command commandMode.
+     *
+     * @throws IllegalValueException if the given command commandMode string is invalid.
+     */
+    public void setCommandMode(String input) throws IllegalValueException {
+        requireNonNull(input);
+        String trimmedInput = input.trim().toLowerCase();
+        if (!isValidMode(trimmedInput)) {
+            throw new IllegalValueException(MESSAGE_COMMANDMODE_CONSTRAINTS);
+        }
+        this.commandMode = trimmedInput;
+    }
+
+    public static boolean isValidMode(String test) {
+        return StringUtil.containsWordIgnoreCase(COMMANDMODE_VALIDATION_REGEX, test);
+    }
+
+    @Override
+    public String toString() {
+        return commandMode;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof CommandMode // instanceof handles nulls
+            && this.commandMode.equals(((CommandMode) other).commandMode)); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return commandMode.hashCode();
+    }
+
+}
+```
+###### /java/seedu/address/model/Model.java
+``` java
+    /**Change the current command mode*/
+    void changeCommandMode(String mode) throws IllegalValueException;
+
+    /**Returns the current command mode*/
+    String getCommandMode();
+```
+###### /java/seedu/address/model/ModelManager.java
+``` java
+    //========== Command Mode ==============================================================================
+    @Override
+    public void changeCommandMode(String mode) throws IllegalValueException {
+        addressBook.changeCommandMode(mode);
+        indicateAddressBookChanged();
+    }
+
+    @Override
+    public String getCommandMode() {
+        return addressBook.getCommandMode();
+    }
+```
+###### /java/seedu/address/model/person/Address.java
+``` java
+    /**
+     * Return the street name of selected person's address
+     */
+    public String getStreetName() {
+        if (value.indexOf("#") < 0) {
+            return value;
+        }
+        return value.substring(0, value.indexOf("#"));
+    }
+```
+###### /java/seedu/address/model/task/TaskContainsKeywordsPredicate.java
+``` java
+/**
+ * Tests that a {@code ReadOnlyTask}'s {@code Description} and/or {@code Deadline} matches any of the keywords given.
+ */
+public class TaskContainsKeywordsPredicate implements Predicate<ReadOnlyTask> {
+    private final List<String> keywords;
+
+    public TaskContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(ReadOnlyTask task) {
+        String testDate = "";
+        String tag = Arrays.toString(task.getTags().toArray())
+                .replaceAll("[\\[\\](),{}]", "");
```

Comment:
Sometimes symbols / regex takes a while for other developer to understand because there is lots of way to do similar things, so you might want to use a string/comment to describe them.